simulations/vip-164/simulations.ts

```diff
@@ -0,0 +1,71 @@
+import { expect } from "chai";
+import { BigNumber } from "ethers";
+import { parseUnits } from "ethers/lib/utils";
+import { ethers } from "hardhat";
+
+import { forking, testVip } from "../../src/vip-framework";
+import {
+  CERTIK_RECEIVER,
+  FAIRYPROOF_RECEIVER,
+  PECKSHIELD_RECEIVER,
+  QUANTSTAMP_RECEIVER,
+  USDC,
+  USDT,
+  vip164,
+} from "../../vips/vip-164";
+import IERC20_ABI from "./abi/IERC20UpgradableAbi.json";
+
+const QUANTSTAMP_VENUS_PRIME_AMOUNT = parseUnits("45000", 18);
+const QUANTSTAMP_INCOME_ALLOCATION_AMOUNT = parseUnits("30000", 18);
+const FAIRYPROOF_AMOUNT = parseUnits("10000", 18);
+const PECKSHIELD_AMOUNT = parseUnits("12000", 18);
+const CERTIK_AMOUNT = parseUnits("19000", 18);
+
+forking(31298100, () => {
+  let usdc: ethers.Contract;
+  let usdt: ethers.Contract;
+  let prevBalanceQuantStamp: BigNumber;
+  let prevBalanceFairyproof: BigNumber;
+  let prevBalancePeckShield: BigNumber;
+  let prevBalanceCertik: BigNumber;
+
+  before(async () => {
+    usdc = new ethers.Contract(USDC, IERC20_ABI, ethers.provider);
+    usdt = new ethers.Contract(USDT, IERC20_ABI, ethers.provider);
+    prevBalanceQuantStamp = await usdc.balanceOf(QUANTSTAMP_RECEIVER);
+    prevBalanceFairyproof = await usdc.balanceOf(FAIRYPROOF_RECEIVER);
+    prevBalancePeckShield = await usdc.balanceOf(PECKSHIELD_RECEIVER);
+    prevBalanceCertik = await usdc.balanceOf(CERTIK_RECEIVER);
+  });
+
+  testVip("VIP-164 Security audits payments", vip164(), {
```

Comment:
We could verify 5 `WithdrawTreasuryBEP20` events are emitted. See [here](https://github.com/VenusProtocol/vips/pull/63/files#diff-0817a963b8a0380e325262d82627dd6056e90de9dcadf19459efaf202dd31619R75)